### ISSUE TITLE
Freecam and Chat while in Inventory

### ISSUE THREAD

**Issue by Atilogit** [ImpactDevelopment/ImpactIssues#2242]
### Is your feature request related to a problem? Please describe.

I was trying to do a dupe which involves two players accessing the inventory of a donkey.
When i tried it on different servers i came across one with the plugin [GriefPrevention](https://github.com/MinecraftPortCentral/GriefPrevention).
It prevents access from two players at the same time by only letting the donkey be accessed by its owner. It is possible to transfer the ownership even when riding the donkey with the use of Freecam (running a command and then rightclicking the donkey through freecam) and still accessing the inventory despite the donkey being owned by another player. Unfortunately because this only works as long as the first player does not unmount the donkey so the better option would be to access the inventory from the outside and then, while the inventory is open transfer the ownership of the donkey to the other player which can then ride the donkey.

**TL;DR**
Being able to run a command and rightclick a donkey while his inventory is open would bypass a plugin blocking a dupe

### Describe the solution you'd like
1. Being able to use chat like similar to what is already present in the Inventory module

2. Being able to enable freecam when in an inventory: When freecam is enabled the inventory gui disappears and the player can move around and click like normal freecam. When it is disabled again the gui reappears and the player can keep using the inventory. (Inventory never gets closed during the process)

### Describe alternatives you've considered
Did not really find anything like this

### Additional context

**Graphical explaination**

Using chat 
![Chat](https://user-images.githubusercontent.com/52137800/80981580-74594700-8e2a-11ea-8815-2f645cff0e91.png)
Using freecam
![Freecam](https://user-images.githubusercontent.com/52137800/80981855-ce5a0c80-8e2a-11ea-81e5-3edb0986394b.png)

**Comment by dominikaaaa** [comment]
Closing this as won't fix, because Impact doesn't update frequently enough to add dupe-specific modules that work for a short period of time.

**Comment by Atilogit** [comment]
I meant to describe it more as a general purpose feature. The specific application for this dupe is just an example showing how useful it could be

**Comment by dominikaaaa** [comment]
You can already enable freecam before you open the inventory.

**Comment by Atilogit** [comment]
Thanks, i didnt know that. But i'm pretty sure its not possible to use chat in an inventory which could be even more useful

**Comment by dominikaaaa** [comment]
You cannot. It wouldn't be that useful as some keys wouldn't even work.

**Comment by Atilogit** [comment]
Couldn't you just paste commands into the console if that happens?

**Comment by dominikaaaa** [comment]
That would still have the same issue.

**Comment by Atilogit** [comment]
Is getting text into the chat really such a big problem that this would be too difficult to implement? I would imagine that there is some other (maybe ugly looking) way to get text input.

**Comment by dominikaaaa** [comment]
The issue is that you would be getting text input.

**Comment by Atilogit** [comment]
I don't really understand the inner works of minecraft but is that not what would be required?

**Comment by dominikaaaa** [comment]
It would cause that issue if you did.

**Comment by dominikaaaa** [comment]
Also you can't open multiple GUIs at once, with some exceptions.

**Comment by Atilogit** [comment]
I see. So its just too much of a hassle to get working properly. Thanks for your explainations at least

**Comment by LeafHacker** [comment]
There's no reason why we couldn't make chat work over inventories, other than the effort of doing it
